### ISSUE TITLE
[Issue]: Not all files in a folder are scanned 

### ISSUE THREAD

**Issue by davipro34** [jellyfin/jellyfin#6828]
### Please describe your bug

In some folders, the library scanner stops at the first entry found (in alphabetical order). The following files are not taken into account, it seems that it does not scan them. The problem does not arise with all files. I couldn't find any logic as to why this happens in some files but not in others.

### Jellyfin Version

10.7.7

### Environment

```markdown
- OS: Debian 10
 - Virtualization: VM guest (KVM Proxmox)
 - Clients: Browser, Android (and app Gelli on Android)
 - Browser: Firefox 91.0.2 (64 bits), Chromium Version 93.0.4577.63 (Build officiel) snap (64 bits)
 - Jellyfin Version: 10.7.6
 - FFmpeg Version: don't know
 - Playback: don't know 
 - Hardware Acceleration: don't know
 - Installed Plugins: AudioDB10.7.6.0 , Auto Organize9.0.0.0 , Kodi Sync Queue6.0.0.0 , MusicBrainz10.7.6.0 , OMDb10.7.6.0, Playback Reporting10.0.0.0 , Reports12.0.0.0
 - Reverse Proxy: none
 - Base URL: none
 - Networking: NAT
 - Storage: local mount of SMB share on NAS
```

### Please attach any screenshots here

![image](https://user-images.githubusercontent.com/78750640/141460866-21393646-6ccb-4c71-8cd9-a5d014eaa9c9.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by davipro34** [comment]
I just tested on a Manjaro OS with a clean and new installation of Jellyfin in version 10.7.7.
Bug # 6828 remains present.
Bug # 6514 remains present.
I have three different installations that have the same issues.
If I need testing I can help.

**Comment by cvium** [comment]
Please include directory/file layout/structure.

**Comment by davipro34** [comment]
I notice this bug is occurring in my music videos (karaoke) folder.
I realize that the scanner creates "versions" in the media library, rather than different clips. But this-only for some files, not all. I do not know why.

At one point I thought it was due to the "ID tags" that were missing on some, but ultimately it is not. Indeed, I have them all "tagged", deleted the media library, then recreated the media library. The problem remains.

In short, why does the scanner create "versions"?

    /mnt/nas1/medias/Karaoke-INT  tree                                                                                                           ✔  8s  
.
├── 00 Divers
│   ├── Anita Ward - Ring My Bell.mp4
│   ├── Anita Ward - Ring My Bell.nfo
│   ├── Anita Ward - Ring My Bell-poster.jpg
│   ├── Bee Gees - Stayin' alive.mp4
│   ├── Bee Gees - Stayin' alive.nfo
│   ├── Bee Gees - Stayin' alive-poster.jpg
│   ├── Donna Summer - Hot stuff.mp4
│   ├── Donna Summer - Hot stuff.nfo
│   ├── Donna Summer - Hot stuff-poster.jpg
│   ├── Gloria Gaynor - I will survive.mp4
│   ├── Gloria Gaynor - I will survive.nfo
│   ├── Gloria Gaynor - I will survive-poster.jpg
│   ├── Kc & The Sunshine Band - That's the way (I like it).mp4
│   ├── Kc & The Sunshine Band - That's the way (I like it).nfo
│   ├── Kc & The Sunshine Band - That's the way (I like it)-poster.jpg
│   ├── Labelle - Lady Marmalade.mp4
│   ├── Labelle - Lady Marmalade.nfo
│   ├── Labelle - Lady Marmalade-poster.jpg
│   ├── Rick Astley - Together Forever.mp4
│   ├── Rick Astley - Together Forever.nfo
│   ├── Rick Astley - Together Forever-poster.jpg
│   ├── The Communards - Don't leave me this way.mp4
│   ├── The Communards - Don't leave me this way.nfo
│   ├── The Communards - Don't leave me this way-poster.jpg
│   ├── The Supermen Lovers - Starlight.mp4
│   ├── The Supermen Lovers - Starlight.nfo
│   └── The Supermen Lovers - Starlight-poster.jpg
├── Earth Wind And Fire
│   ├── Earth Wind And Fire - After The Love Has Gone.mp4
│   ├── Earth Wind And Fire - After The Love Has Gone.nfo
│   ├── Earth Wind And Fire - Boogie Wonderland.mp4
│   ├── Earth Wind And Fire - Fantasy - Karaoke Version from Zoom Karaoke.mp4
│   ├── Earth Wind And Fire - Let's Groove.mp4
│   ├── Earth Wind And Fire - September.mp4
│   └── folder.jpg
├── Imagination
│   ├── folder.jpg
│   ├── Imagination - Body Talk.mp4
│   ├── Imagination - Body Talk.nfo
│   ├── Imagination - Flashback.mp4
│   ├── Imagination - Just an illusion.mp4
│   └── Imagination - Music and lights.mp4
├── Kool and the gang
│   ├── folder.jpg
│   ├── Kool and the gang - Celebration.mp4
│   ├── Kool and the gang - Celebration.nfo
│   ├── Kool and the gang - Cherish.mp4
│   ├── Kool and the gang - Fresh.mp4
│   ├── Kool and the gang - Get down on it.mp4
│   ├── Kool and the gang - Joanna.mp4
│   ├── Kool and the gang - Ladies Night.mp4
│   ├── Kool and the gang - Let's go dancin.mp4
│   ├── Kool and the gang - Take my heart.mp4
│   ├── Kool and the gang - Tonight.mp4
│   └── Kool and the gang - Too Hot.mp4
└── Lionel Richie
    ├── Lionel Richie - All night long.mp4
    ├── Lionel Richie - All night long.nfo
    ├── Lionel Richie - All night long-poster.jpg
    ├── Lionel Richie - Ballerina girl.mp4
    ├── Lionel Richie - Ballerina girl.nfo
    ├── Lionel Richie - Ballerina girl-poster.jpg
    ├── Lionel Richie - Dancing on the ceiling.mp4
    ├── Lionel Richie - Dancing on the ceiling.nfo
    ├── Lionel Richie - Dancing on the ceiling-poster.jpg
    ├── Lionel Richie Diana Ross - Endless Love.mp4
    ├── Lionel Richie Diana Ross - Endless Love.nfo
    ├── Lionel Richie Diana Ross - Endless Love-poster.jpg
    ├── Lionel Richie - Do it to me.mp4
    ├── Lionel Richie - Do it to me.nfo
    ├── Lionel Richie - Do it to me-poster.jpg
    ├── Lionel Richie - Hello.mp4
    ├── Lionel Richie - Hello.nfo
    ├── Lionel Richie - Hello-poster.jpg
    ├── Lionel Richie - Penny Lover.mp4
    ├── Lionel Richie - Penny Lover.nfo
    ├── Lionel Richie - Penny Lover-poster.jpg
    ├── Lionel Richie - Running with the night.mp4
    ├── Lionel Richie - Running with the night.nfo
    ├── Lionel Richie - Running with the night-poster.jpg
    ├── Lionel Richie - Say You, Say Me.mp4
    ├── Lionel Richie - Say You, Say Me.nfo
    ├── Lionel Richie - Say You, Say Me-poster.jpg
    ├── Lionel Richie - Still.mp4
    ├── Lionel Richie - Still.nfo
    ├── Lionel Richie - Still-poster.jpg
    ├── Lionel Richie - Stuck on you.mp4
    ├── Lionel Richie - Stuck on you.nfo
    ├── Lionel Richie - Stuck on you-poster.jpg
    ├── Lionel Richie The Commodores - Easy.mp4
    ├── Lionel Richie The Commodores - Easy.nfo
    ├── Lionel Richie The Commodores - Easy-poster.jpg
    ├── Lionel Richie The Commodores - Sail on .mp4
    ├── Lionel Richie The Commodores - Sail on .nfo
    ├── Lionel Richie The Commodores - Sail on -poster.jpg
    ├── Lionel Richie - Three times a lady .mp4
    ├── Lionel Richie - Three times a lady .nfo
    ├── Lionel Richie - Three times a lady -poster.jpg
    ├── Lionel Richie - Truly.mp4
    ├── Lionel Richie - Truly.nfo
    └── Lionel Richie - Truly-poster.jpg

5 directories, 97 files

**Comment by davipro34** [comment]
Above we see that,
For the "Lionel Richie" and "00 Divers" files, we have a file with several titles that have been scanned;
While for the files "Earth Wind And Fire", "Imagination" and "Kool and the gang" only the first title in alphabetical order has been "scanned".

**Comment by davipro34** [comment]
![image](https://user-images.githubusercontent.com/78750640/141988070-364f9123-6e1b-4195-937c-c884d1b50182.png)

**Comment by davipro34** [comment]
![image](https://user-images.githubusercontent.com/78750640/141988321-41cc9c7b-6774-4267-8f0d-e2314a79d019.png)

**Comment by crobibero** [comment]
What type of library did you select for your music videos?

**Comment by davipro34** [comment]
On this screenshot, we seeing well the difference : 

![image](https://user-images.githubusercontent.com/78750640/141988512-186a8e3b-5554-4969-bd8b-c830bed58a9b.png)

**Comment by davipro34** [comment]
Hello,
I tried to use "Music Videos" and "Other". But the result is the same.

**Comment by davipro34** [comment]
I tried to use the "Music" type, no title appears in the library.

**Comment by davipro34** [comment]
I tried to use the "Movies" type. But the result is the same.

**Comment by nodje** [comment]
It seems my issue is the same as this one #5949
And the behaviour description here is better than what I reported: it's not so much that "folders" are missing, but yeah, it seems the scanner stops at the first item it encounters.
But not systematically unfortunately. So it's hard to provide steps to reproduce.

All I can say it moving the missing items one folder up solves the issue.
But again, since you don't know which folders haven't been scanned, it's impossible to know what to do to get all the library content scanned successfully.

To illustrate what is wrong:
```
|---Directors
|-----China
|-----Sweden
|-----USA
```
are all found in the top level folder `Directors` added to a Movie type library.
But subfolders of `China` and `Sweden` aren't found, even though all subfolders of `USA` ARE found (as much as I can see)

If I move `China` a level up and add it a to a Library of its own, no problem, all content is found.
If I put it back to `Directors`, `China` isn't even found.

This is a big deal, this issue should be given priority.
Is there anything in 10.8.0 potentially dealing with this?

**Comment by nodje** [comment]
Ok I see a pattern now:
each time there's a single file in a folder hierarchy, the subfolders at the same level are not taken into account.

In my example, this means:
```
|---Directors
|-----China
|-------movie1.mkv
|-------Movie2 Folder
|---------movie2.mkv
```
`Movie2 Folder` is never to be scanned by the Movie Library
I can solve this issue by making sure `movie1.mkv` is put in a folder.

That being said, I'm not sure how this would behave for movies and folders down a level.
I would expect logically the same behaviour. "allowed" folders like `featurettes` or the likes are found though.

**Comment by nodje** [comment]
More on this: playing with the `Folder` mode within the added Library made me realize a basic expectation form Jellyfin.

Afaiu, a folder which contains one media file is considered a top folder for a movie (potentially music or serie), and anything within it should be concerning that movie (see the list of conventionally accepted forlder `extras`,`featurettes` etc.).
A folder with only sub-folders, is a "normal" folder.

And as such if you put multiple media file in a folder, they are wrongly considered "one media" in Folders. (I'm not sure how this reflects in the Library itself, probably all media are identified independently).

I'm not sure whether this is stated clearly in the doc but it makes the requirements for a OK media hierarchy quite difficult to meet. (I'll have a LOT of work to make each of my libs answer this req.)

I'm not sure how difficult it'd be for the scanner to be able to differentiate "movie folder" from "mix-bag movie folder movie file" folders, but it's needed ASAP IMHO. And along with a clear documentation.

**Comment by Shadowghost** [comment]
**TLDR:** None of the here mentioned filesystem layouts adhere to any of our officially supported schemas. The docs are quite clear about how you should structure your library.

@davipro34  you are getting multiple versions of the same movie because your files are named as such (the hyphen is the idicator for a version in your case): Check the [docs](https://jellyfin.org/docs/general/server/media/movies.html#multiple-versions-of-a-movie) on how you can prevent that.

@nodje you have a library with a depth more than one directory - which is not explicitly unsupported, but the [docs](https://jellyfin.org/docs/general/server/media/movies.html) are quite clear about how you should structure your media for proper pickup.
Your current layout should be handled properly without any modification from your side if you add the `China`, `Sweden` and `USA` folders as seperate source folders to your movie library (you can define multiple media sources per library).

----
It's not an unknown fact that the scanner is quite picky but it's not an easy task to fix it. If you want is ASAP, you need to do it yourself since there are more pressing matters the team is working on. It will eventually be worked on cleaning up the scanner and (hopefully) making it more robust after the efcore database migration is done (not to be finished and fully integrated in the near future).

**Comment by sunnysamantara** [comment]
Does anyone know on what version of jellyfin this issue is not happening, I'm trying for the last 2 months and facing the same issue, or is there any workaround to bypass this

**Comment by sevenrats** [comment]
following the documented naming structure to avoid these issues
